src/unitxt/inference.py

```diff
@@ -106,6 +121,7 @@ def _infer(self, dataset):
         return ["[[10]]" for instance in dataset]
 
 
+@deprecation(version="1.11.0", alternative="Plain dictionary")
```

Comment:
This interface does not contain all parameters and it's errorprone because it has to be updated with every change in GenAI sdk. We can just use plain dict instead and validate it using the `TextGenerationParameters` model from SDK. I suggest a similar approach with other models (openAI parameters)

Reply:
The problem is most user don't know what the params are. The definition is for the user who do not know what their names are. I prefer it to be fully defined and set the version of the sdk.

Reply:
Yes.   It's a tradeoff :
   type API (that is self including documentation) - that needs to be updated with each version of SDK
vs 
   non typed  API - that requires the user to go to the external API to see the params and assume it's compatible
   
@OfirArviv - Not that today - we don't require specific version of any dependent  library in unitxt, so people can choose whatever fit for them given their other requirements - we assume the level of API we use does not change.   This is a tradeoff in itself - fexlibility for user, vs risk on mismatch between version.

Reply:
Would it be better if the Artifact class is dynamically generated from the pydantic type (I can look into this) or do you still prefer to have it explicitly written in code so that people can inspect it?

Reply:
I think at this point, I'd prefer explicating typing, so it's clear documented.   This helps the users understand what they can and can not control.    I realize it's not a clear cut case, but it's more consistent with other places in the library.